### ISSUE TITLE
Structurally singular system

### ISSUE THREAD

**Issue by tbeu** [modelica/Modelica_DeviceDrivers#122]
Reusing a (serial) package for multiple senders results in a structurally singular system.

![ss](https://cloud.githubusercontent.com/assets/5444067/16301739/5e6a8a30-3947-11e6-9594-b1439b8d3b94.png)

**Comment by bernhard-thiele** [comment]
The `PackageIn` and `PackageOut` connectors have both, inputs and outputs.
Hence, there is also a backward signal flow and this is why the example above doesn't work.

**Comment by tbeu** [comment]
Is it a current limitation of the SerialPackager we need to accept or is there a workaround. I consider this limitation as rather critical.

**Comment by bernhard-thiele** [comment]
This is a limitation and I don't see a simple workaround. One could provide a distinguished splitter block, but one needs think how to handle the back propagation. Also, once one can split into parallel branches, somebody could be tempted to add packages in parallel branches.

Another workaround could be to have a feedthrough at a send block which allows to connect another send block in series. Not sure what is more feasible.

**Comment by tbeu** [comment]
Since package branching is not allowed anywhere, I would prefer a splitter block.

Do you know of any (tool-specific) way to prohibit branching, i.e. have max (exactely) 1 connection per connector? We could use cardinality (like in MSL Fluid) but it is deprecated.

**Comment by tbeu** [comment]
https://trac.modelica.org/Modelica/ticket/1410 proposes to use cardianality only in asserts. Thus we could simply do that.

**Comment by bernhard-thiele** [comment]
I have no idea how one could prohibit this and I also have no clear picture how to implement. Probably it needs a trial and error approach.

**Comment by tbeu** [comment]
Adding an `assert(cardinality(..))` for each element with PackageIn and PackageOut connectors could help. What do you think?

**Comment by bernhard-thiele** [comment]
For me it seems that in this way one maybe could get better error messages for systems that branch out, but it still wouldn't allow you to have several sinks as you would like to have (or am I missing something)?

**Comment by tbeu** [comment]
Yes, instead of the singular sysem error msg you would get the cardinality assert.

Parallel branches are simple not possible. I beleive a user needs to define its own block if he/she wants to model multiple sinks.

**Comment by tbeu** [comment]
Too bad, adding a cardinality assert message does not help the user since the error message on structurally singularity is raised before the assert (in SimulationX).
